### DISCUSSION TITLE
Support validateOnly without relying on context of rules() method on component

### DISCUSSION THREAD

**Discussion by penagos** [livewire/livewire#2577] [3. Feature ideas]
It would be nice if we could validate a subset of properties on a component via validateOnly/validate orthogonal to the rules defined in the `rules` method on a component. Consider the following code snippet:
```
class MyComponent extends Component {
    public $baz;
    public $bar;
    public $bat;

    public function foo()
    {
        // Update some model state based on 2/3 required props only
        $validatedData = $this->validateOnly['baz, bar'], [
            'baz' => 'required',
            'bar' => 'required'
        ]);
    }
    public function submit()
    {
        // Here we only care about the bat property being in the request
        $data = $this->validate();
    }
    public function rules()
    {
        return [
            'bat' => 'required'
        ];
    }
}
```

Instead, it seems to achieve such behavior I need to explicitly call validate with the subset of rules I'm interested in:
```
public function foo()
{
    // This will error
    $data = $this->validate([
        'baz' => 'required',
        'bar' => 'required'
    ]);
}
```
However, it seems property validation still searches for these rules on the rules() method of the component, resulting in:
```
Cannot bind property <property_name> without a validation rule present in the [$rules] array on Livewire component: <component>. 
```

I can of course, circumnavigate the issue by making `bar` and `baz` nullable in the `rules()` method:
```
    public function rules()
    {
        return [
            'bat' => 'required',
            'bar' => 'nullable',
            'baz' => 'nullable'
        ];
    }
```
But in my opinion, that doesn't make it immediately clear whether or not `bar` and `baz` are required?

**Comment by joshhanley** [comment]
@penagos yep agreed, ran into a similar thing today. I think we should be able to pass an array into validateOnly and have it use the existing rules.

If I get a chance I will look into it.

**Reply by penagos** [reply]
@joshhanley out of curiosity is this being actively looked at? If not, are you open to PR requests?

**Reply by joshhanley** [reply]
@penagos I haven't had a chance to look at it yet. Caleb is open to PRs, so if you submit one, I will happily support it 🙂
